lspSmithy/src/org/scala/abusers/lspsmithy/SmithyConverter.scala

```diff
@@ -212,22 +198,7 @@ object SmithyConverter:
 
       case StructureLiteralType(StructureLiteral(properties, _)) =>
         val id = uniqueShapeId("InlineStruct")
-        properties
-          .traverse { prop =>
-            val memberId = id.withMember(prop.name.value)
-            smithyType(prop.tpe, namespace).map { target =>
-              val member = MemberShape
-                .builder()
-                .id(memberId)
-                .target(target)
-              val memberOpt =
-                if prop.optional.no then
-                  member.addTrait(new RequiredTrait.Provider().createTrait(RequiredTrait.ID, Node.objectNode))
-                else member
-
-              memberOpt.build()
-            }
-          }
+        structureMembers(id, properties)
           .map(_.foldLeft(StructureShape.builder().id(id)) { case (acc, item) => acc.addMember(item) }.build())
           .flatMap { structureShape =>
             State(shapes => (shapes + (id -> structureShape), id))
```

Comment:
nitpick: imo the state shouldn't be a `Map` - the `Shape` contains an ID anyway, so you could just keep a Set of shapes and you could still turn it into a map internally. At the very least the modifications to `State` shouldn't require passing both the id and shape, as it risks using the wrong id accidentally

Reply:
good point. I think I wanted initially to check if we are not generating multiple different shapes with the same id. But we are not checking it so it could be simplified.